homeassistant/components/android_ip_webcam/__init__.py

```diff
@@ -20,334 +19,118 @@
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=entry
             )
         )
 
-        if sensors:
-            hass.async_create_task(
-                discovery.async_load_platform(
-                    hass,
-                    Platform.SENSOR,
-                    DOMAIN,
-                    {CONF_NAME: name, CONF_HOST: host, CONF_SENSORS: sensors},
-                    config,
-                )
-            )
-
-        if switches:
-            hass.async_create_task(
-                discovery.async_load_platform(
-                    hass,
-                    Platform.SWITCH,
-                    DOMAIN,
-                    {CONF_NAME: name, CONF_HOST: host, CONF_SWITCHES: switches},
-                    config,
-                )
-            )
-
-        if motion:
-            hass.async_create_task(
-                discovery.async_load_platform(
-                    hass,
-                    Platform.BINARY_SENSOR,
-                    DOMAIN,
-                    {CONF_HOST: host, CONF_NAME: name},
-                    config,
-                )
-            )
-
-    tasks = [async_setup_ipcamera(conf) for conf in config[DOMAIN]]
-    if tasks:
-        await asyncio.wait(tasks)
-
     return True
 
 
-class AndroidIPCamEntity(Entity):
-    """The Android device running IP Webcam."""
-
-    def __init__(self, host, ipcam):
-        """Initialize the data object."""
-        self._host = host
-        self._ipcam = ipcam
-
-    async def async_added_to_hass(self):
-        """Register update dispatcher."""
-
-        @callback
-        def async_ipcam_update(host):
-            """Update callback."""
-            if self._host != host:
-                return
-            self.async_schedule_update_ha_state(True)
-
-        self.async_on_remove(
-            async_dispatcher_connect(self.hass, SIGNAL_UPDATE_DATA, async_ipcam_update)
-        )
-
-    @property
-    def should_poll(self):
-        """Return True if entity has to be polled for state."""
-        return False
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Android IP Webcam from a config entry."""
+    websession = async_get_clientsession(hass)
+    cam = PyDroidIPCam(
+        websession,
+        entry.data[CONF_HOST],
+        entry.data[CONF_PORT],
+        username=entry.data.get(CONF_USERNAME),
+        password=entry.data.get(CONF_PASSWORD),
+        timeout=entry.data[CONF_TIMEOUT],
+        ssl=False,
+    )
+    coordinator = AndroidIPCamDataUpdateCoordinator(hass, entry, cam)
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
-    @property
-    def available(self):
-        """Return True if entity is available."""
-        return self._ipcam.available
+    return True
 
-    @property
-    def extra_state_attributes(self):
```

Comment:
Removing the extra attributes is a breaking change that we need to describe in the breaking change paragraph.